### ISSUE TITLE
[@typescript-eslint/ban-types] Rule text includes newline, breaks VSCode problem matcher

### ISSUE THREAD

**Issue by thw0rted** [typescript-eslint/typescript-eslint#2895]
- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
    "extends": [
        "eslint:recommended",
        "plugin:@typescript-eslint/recommended"
    ],
}
```

```TS
function foo<T extends object>(f: T) {
 // do something
}
```

(I don't think my `tsconfig.json` is relevant to this issue.)

**Expected Result**

The entire rule message should be written with no newlines so that the problem matcher identifies it as a single violation.  (I think this is supposed to be a requirement for writing `eslint` rules but I'm having trouble finding documentation about how to write a compliant rule, as opposed to documentation about rules that validate newlines in code.)

**Actual Result**

The rule message looks like

````text
  68:39  error    Don't use `object` as a type. The `object` type is currently hard to use ([see this issue](https://github.com/microsoft/TypeScript/issues/21732)).
Consider using `Record<string, unknown>` instead, as it allows you to more easily inspect and use the keys  @typescript-eslint/ban-types
````

Note that there is a newline inserted immediately before the word "Consider".  There are other violation messages for this rule that have the same problem, such as

````text
Don't use `Function` as a type. The `Function` type accepts any function-like value.
It provides no type safety when calling the function, which can be a common source of bugs.
It also accepts things like class declarations, which will throw at runtime as they will not be called with `new`.
If you are expecting the function to accept certain arguments, you should explicitly define the function shape
````

where each sentence is preceded by a newline character.  The "Problems" tab in VSCode parses this as a file called "ban-types" (preceded by a "path" consisting of the last sentence of the error message), then flags each subsequent violation as an error on that (non-existent) file.

As far as I can tell, the `ban-types` rule is the only rule that runs under the `recommended` rule set that outputs multi-line messages like this.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.11.0` |
| `@typescript-eslint/parser`        | `4.10.0` |
| `TypeScript`                       | `4.1.3` |
| `ESLint`                           | `7.15.0` |
| `node`                             | `12.7.0` |

**Comment by bradzacher** [comment]
It's completely valid for lint rules to output messages with newlines (or any tooling for that matter)

Consider raising an issue with VSCode.
